tests/upgrades/test_capsule.py

```diff
@@ -223,3 +224,71 @@ def test_post_user_scenario_capsule_sync_2(self):
             host=self.cap_host
         )[self.cap_host]
         self.assertEqual('0', result)
+
+
+class Scenario_capsule_sync_3(APITestCase):
+    """The test class contains a post-upgrade scenario to test if puppet repo added to
+    satellite is synced to External capsule.
+
+    Test Steps:
+
+        1. Upgrade Satellite and Capsule.
+        2. Sync a puppet repo, Add to C.V. in satellite.
+        3. Run capsule sync and check if it's completed successfully.
+
+    BZ: 1746589
+    """
+    @classmethod
+    def setUpClass(cls):
+        cls.cap_host = settings.upgrade.rhev_cap_host or settings.upgrade.capsule_hostname
+        cls.org = entities.Organization().search(
+            query={'search': 'name="{}"'.format(DEFAULT_ORG)})[0]
+        cls.lc_env = entities.LifecycleEnvironment(organization=cls.org).search(
+            query={'search': 'name="{}"'.format('Dev')})[0]
+
+    @post_upgrade
+    def test_post_user_scenario_capsule_sync_3(self):
+        """ Post-upgrade scenario that creates and sync repository with
```

Comment:
Any reason for keeping this in a separate class?

Reply:
Yeah .. to keep separate env for setup before execute the post upgrade test.